android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -113,10 +133,37 @@ constructor(open val fhirEngine: FhirEngine, open val dispatcherProvider: Dispat
 
   suspend fun getBinary(id: String): Binary = fhirEngine.get(id)
 
-  suspend fun save(resource: Resource) {
+  suspend fun create(vararg resource: Resource): List<String> {
+    val syncStrategy = mutableMapOf<String, List<String>>()
+
+    sharedPreferencesHelper.read<List<String>>(
+        key = SharedPreferenceKey.PRACTITIONER_DETAILS_CARE_TEAM_IDS.name
+      )
+      ?.let { careTeamIds -> syncStrategy[SyncStrategy.CARE_TEAM.value] = careTeamIds }
+
+    sharedPreferencesHelper.read<List<String>>(
+        key = SharedPreferenceKey.PRACTITIONER_DETAILS_ORGANIZATION_IDS.name
+      )
+      ?.let { organizationIds -> syncStrategy[SyncStrategy.ORGANIZATION.value] = organizationIds }
+
+    sharedPreferencesHelper.read<List<String>>(
+        key = SharedPreferenceKey.PRACTITIONER_DETAILS_LOCATION_IDS.name
+      )
+      ?.let { locationIds -> syncStrategy[SyncStrategy.LOCATION.value] = locationIds }
+
+    sharedPreferencesHelper.read<KeycloakUserDetails>(
+        key = SharedPreferenceKey.PRACTITIONER_DETAILS_USER_DETAIL.name
+      )
+      ?.let { practitioner ->
+        syncStrategy[SyncStrategy.PRACTITIONER.value] = listOf(practitioner.id)
+      }
+
     return withContext(dispatcherProvider.io()) {
-      resource.generateMissingId()
-      fhirEngine.create(resource)
+      resource.map {
+        it.generateMissingId()
+        it.addMandatoryTags(syncStrategy)
+      }
```

Comment:
I think the kotlin scope function `apply` is more appropriate here than lambda `map`
```suggestion
      resource.apply {
        generateMissingId()
        addMandatoryTags(syncStrategy)
      }
```

Reply:
Curious.
In this code, `resource` is a vararg.

Is it still appropriate to use `.apply` ?

Reply:
Oh I see.

Reply:
How about `onEach` lambda is it possible to use here?

Reply:
Because I think we are updating each item here on the `varags`

Reply:
Yeah, sure